wordle-web/test/wordOptionTest.test.ts

```diff
@@ -0,0 +1,66 @@
+import { wordOptions } from '~/scripts/wordOptions'
+import { WordsService } from '~/scripts/wordsService'
+
+
+
+ describe('AvailableWord', () => {
+   let word
+   let words
+   let allWords
+   test('is an instance', () => {
+     words = new wordOptions()
+     expect(words).toBeTruthy()
+   })
+
+   test('all words found', () => {
+     word = '?????'
+     words = new wordOptions()
+     allWords = words.validWords(word)
+     expect(allWords.length).toBe(WordsService.getWords().length)
+   })
+
+   test('no words found', () => {
+     words = new wordOptions()
+     expect(words.count).toBe(0)
+     word = 'APPLE'
+     allWords = words.validWords(word)
+     expect(words.count).toBe(0)
+     expect(allWords[0]).toBe('')
+     expect(allWords[1]).toBeFalsy()
+   })
+
+   test('word A????', () => {
+     words = new wordOptions()
+     word = 'A????'
+     allWords = words.validWords(word)
+     expect(allWords.length).toBeGreaterThan(0)
+     expect(allWords.length).toBe(29)
+     expect(words.count).toBe(29)
+     expect(allWords.includes('ANGEL')).toBeTruthy()
+     expect(allWords.includes('BATON')).toBeFalsy()
+   })
```

Comment:
As mentioned above, even with the regex working properly, some tests were failing.

This is because all of the strings are lower case in our list of words! Simply fixing the casing on your tests was enough to get them to pass after implementing the change I suggested to your wordOptions logic.